src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp

```diff
@@ -317,50 +317,32 @@ std::string EvaluableNodeTreeManipulation::MixStrings(const std::string &a, cons
 	return result;
 }
 
-bool EvaluableNodeTreeManipulation::DoesTreeContainLabels(EvaluableNode *en)
-{
-	if(en == nullptr)
-		return false;
-
-	if(en->GetNumChildNodes() == 0)
-		return (en->GetNumLabels() > 0);
-
-	if(!en->GetNeedCycleCheck())
-		return NonCycleDoesTreeContainLabels(en);
-
-	EvaluableNode::ReferenceSetType checked;
-	return DoesTreeContainLabels(en, checked);
-}
-
 std::pair<EvaluableNode::LabelsAssocType, bool> EvaluableNodeTreeManipulation::RetrieveLabelIndexesFromTreeAndNormalize(EvaluableNode *en)
 {
 	EvaluableNode::LabelsAssocType index;
-	EvaluableNode::ReferenceSetType checked;
-
-	//can check faster if don't need to check for cycles
-	bool en_cycle_free = (en == nullptr || !en->GetNeedCycleCheck());
-	bool label_collision = CollectLabelIndexesFromNormalTree(en, index, en_cycle_free ? nullptr : &checked);
+	if(en == nullptr)
+		return std::make_pair(index, true);
 
 	//if no collision, return
-	if(!label_collision)
-		return std::make_pair(index, false);
+	EvaluableNode::ReferenceSetType checked;
+	if(CollectLabelIndexesFromTree(en, index, en->GetNeedCycleCheck() ? &checked : nullptr))
+		return std::make_pair(index, true);
 
 	//keep replacing until don't need to replace anymore
 	EvaluableNode *to_replace = nullptr;
 	while(true)
 	{
 		index.clear();
 		checked.clear();
-		bool replacement = CollectLabelIndexesFromTreeAndMakeLabelNormalizationPass(en, index, checked, to_replace);
 
-		if(!replacement)
+		if(CollectLabelIndexesFromTreeAndMakeLabelNormalizationPass(en, index, checked, to_replace))
 			break;
 	}
 
 	//things have been replaced, so anything might need to be updated
 	EvaluableNodeManager::UpdateFlagsForNodeTree(en, checked);
 
-	return std::make_pair(index, true);
+	return std::make_pair(index, false);
```

Comment:
This function was returning false before (here and line 329), but now it returns true in these cases. Just want to double check that is indeed correct. I know some of the checks flipped (Cycle) so this might be the same case at the calling sights.

Reply:
I believe this is correct, needed to verify myself.